Controller/ResourceController.php

```diff
@@ -32,21 +34,34 @@ public function __construct(
 
     public function resourceAction($repositoryName, $path)
     {
-        $repository = $this->registry->get($repositoryName);
-        $resource = $repository->get('/' . $path);
-
-        $context = SerializationContext::create();
-        $context->enableMaxDepthChecks();
-        $context->setSerializeNull(true);
-        $json = $this->serializer->serialize(
-            $resource,
-            'json',
-            $context
-        );
-
-        $response = new Response($json);
-        $response->headers->set('Content-Type', 'application/json');
-        
+        try {
+            $repository = $this->registry->get($repositoryName);
+            $resource = $repository->get('/' . $path);
+
+            $context = SerializationContext::create();
+            $context->enableMaxDepthChecks();
+            $context->setSerializeNull(true);
+            $json = $this->serializer->serialize(
+                $resource,
+                'json',
+                $context
+            );
+
+            $response = new Response($json);
+            $response->headers->set('Content-Type', 'application/json');
+        } catch (\Exception $e) {
```

Comment:
I think @dantleech wanted to avoid a dependency on FOSRestBundle .. but maybe it would be a good idea to count the code duplication and limitations caused by not depending on FOSRestBundle and then deciding on it once we have a tally :)